src/app/credExplorer/markovChain.test.js

```diff
@@ -120,3 +122,63 @@ describe("sparseMarkovChainAction", () => {
     expect(pi1).toEqual(expected);
   });
 });
+
+function expectAllClose(
+  actual: Float64Array,
+  expected: Float64Array,
+  epsilon: number = 1e-6
+): void {
+  expect(actual).toHaveLength(expected.length);
+  for (let i = 0; i < expected.length; i++) {
+    if (Math.abs(actual[i] - expected[i]) >= epsilon) {
+      expect(actual).toEqual(expected); // will fail
+      return;
+    }
+  }
+  // Trigger an assertion anyway (for, e.g., `expect.hasAssertions()`).
+  expect(actual).toEqual(actual);
```

Comment:
I'm curious for more context on this. Don't we already have an assertion up on line 131?

Reply:
Good point. This line is indeed not necessary.